subprojects/core/src/main/java/org/gradle/internal/classpath/InstrumentingClasspathFileTransformer.java

```diff
@@ -112,19 +112,28 @@ public InstrumentingClasspathFileTransformer(
         FileLockManager fileLockManager,
         ClasspathWalker classpathWalker,
         ClasspathBuilder classpathBuilder,
+        ClasspathFileHasher classpathFileHasher,
         Policy policy,
         CachedClasspathTransformer.Transform transform,
         GradleCoreInstrumentingTypeRegistry gradleCoreInstrumentingTypeRegistry
     ) {
         this.fileLockManager = fileLockManager;
         this.classpathWalker = classpathWalker;
         this.classpathBuilder = classpathBuilder;
+
+        HashCode configHash = configHashFor(policy, transform, gradleCoreInstrumentingTypeRegistry);
+        this.fileHasher = sourceSnapshot -> {
+            Hasher hasher = Hashing.defaultFunction().newHasher();
```

Comment:
💅 Extract this block to a method to keep the constructor simple and mainly about assigning fields